packages/headless/src/ssr-next/search/engine/search-engine.ssr.ts

```diff
@@ -101,64 +106,62 @@ export function defineSearchEngine<
   >,
 >(options: SearchEngineDefinitionOptions<TControllerDefinitions>) {
   const {controllers: controllerDefinitions, ...engineOptions} = options;
-  type BuildFunction = EngineBuildResult<
-    SSRSearchEngine,
-    TControllerDefinitions,
-    SearchEngineOptions
-  >;
-  type Definition = SearchEngineDefinition<TControllerDefinitions>;
-  type FetchStaticStateFunction = Definition['fetchStaticState'];
-  type HydrateStaticStateFunction = Definition['hydrateStaticState'];
-  type BuildParameters = Parameters<BuildFunction>;
-  type FetchStaticStateParameters = Parameters<FetchStaticStateFunction>;
-  type HydrateStaticStateParameters = Parameters<HydrateStaticStateFunction>;
-
-  const getOptions = () => {
-    return engineOptions;
-  };
 
-  const setNavigatorContextProvider = (
-    navigatorContextProvider: NavigatorContextProvider
-  ) => {
-    engineOptions.navigatorContextProvider = navigatorContextProvider;
+  // biome-ignore lint/suspicious/noExplicitAny: Complex parameter typing would require extensive refactoring
+  const hydrateStaticState = async (...params: any[]) => {
+    const engine = buildSSRSearchEngine(engineOptions);
+    const controllers = buildControllerDefinitions({
+      definitionsMap: (controllerDefinitions ?? {}) as TControllerDefinitions,
+      engine,
+      propsMap:
+        {} as InferControllerPropsMapFromDefinitions<TControllerDefinitions>,
+    });
+
+    const [staticState] = params;
+    engine.dispatch(staticState.searchAction);
+    await engine.waitForSearchCompletedAction();
+    return {engine, controllers};
   };
 
-  const build: BuildFunction = async (...[buildOptions]: BuildParameters) => {
-    const logger = buildLogger(options.loggerOptions);
-    if (!getOptions().navigatorContextProvider) {
-      logger.warn(
-        '[WARNING] Missing navigator context in server-side code. Make sure to set it with `setNavigatorContextProvider` before calling fetchStaticState()'
-      );
-    }
-    const engine = buildSSRSearchEngine(
-      buildOptions?.extend
-        ? await buildOptions.extend(getOptions())
-        : getOptions()
-    );
+  // biome-ignore lint/suspicious/noExplicitAny: Parameter typing matches the interface contract
+  const fetchStaticState = async (...params: any[]) => {
```

Comment:
Shouldn't we change the definition to something like that instead of having to biome-ignore it, and then cast it twice ? Would it be considered a breaking change? 

```typescript
...param: unknown[]
```